src/argilla/server/apis/v1/handlers/oauth2.py

```diff
@@ -0,0 +1,96 @@
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from fastapi import APIRouter, Depends, HTTPException, Request
+from fastapi.responses import RedirectResponse
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from argilla.server.contexts import accounts
+from argilla.server.database import get_async_db
+from argilla.server.enums import UserRole
+from argilla.server.errors.future import AuthenticationError
+from argilla.server.models import User
+from argilla.server.schemas.v1.oauth2 import Provider, Providers, Token
+from argilla.server.security.authentication.jwt import JWT
+from argilla.server.security.authentication.oauth2 import OAuth2ClientProvider
+from argilla.server.security.authentication.userinfo import UserInfo
+from argilla.server.security.settings import settings
+
+router = APIRouter(prefix="/oauth2", tags=["Authentication"])
+
+
+_USER_ROLE_ON_CREATION = UserRole.annotator
+
+
+@router.get("/providers")
+def list_providers(_: Request) -> Providers:
+    items = [Provider(name=provider_name) for provider_name in settings.oauth.providers]
+
+    return Providers(items=items)
+
+
+@router.get("/providers/{provider}/authentication")
+def get_authentication(request: Request, provider: str) -> RedirectResponse:
+    _check_oauth_enabled_or_raise()
+
+    provider = _get_provider_by_name_or_raise(provider)
+    return provider.authorization_redirect(request)
+
+
+@router.get("/providers/{provider}/access-token")
+async def get_access_token(request: Request, provider: str, db: AsyncSession = Depends(get_async_db)) -> Token:
+    _check_oauth_enabled_or_raise()
+
+    try:
+        provider = _get_provider_by_name_or_raise(provider)
+        current_user = UserInfo(await provider.get_user_data(request))
```

Comment:
Maybe change `current_user` to `user_info` instead so we don't think it's an actual user:

```suggestion
        user_info = UserInfo(await provider.get_user_data(request))
```